test/secmemtest.c

```diff
@@ -7,13 +7,15 @@
  * https://www.openssl.org/source/license.html
  */
 
+#include "../e_os.h"  /* pull in unistd.h for _POSIX_VERSION */
```

Comment:
I view this, dependency on private header outside libcrypto, as inadequate idea. [Even plain inappropriate in apps/\*, which should rely exclusively on public headers.] In other words I reckon that we should be striving to *minimize* its usage, not to throw in more of it...

Reply:
There are already a number of those tests including "../e_os.h" (and one "e_os.h")
If you have a better idea, how to get "unistd.h" included without naming it?

Reply:
> There are already a number of those tests including "../e_os.h"

Yes, and suggestion is to stop digging hole deeper.

> If you have a better idea, how to get "unistd.h" included without naming it?

As already discussed in this thread it should be detectable at run-time. [And be platform-neutral, detection that is.] I'm not intimately familiar with mem_sec.c, so can't make good suggestion. @richsalz should be able.

Reply:
We can leave this test as it is, thus it will fail dependent on POSIX version and configuration.
If that is generally considered to be a better idea.